### ISSUE TITLE
Unclear how to write a component that takes either `Store<T>` or `Field<T>`

### ISSUE THREAD

**Issue by wrl** [leptos-rs/leptos#3102]
revisiting the code from #3086:

i have these two components:
```rust
#[component]
fn SubItemListView(
    items: KeyedSubfield<Field<Item>, Item, Uuid, Vec<Item>>
) -> impl IntoView
{
    view! {
        <ul class="items">
            <For
                each = move || items
                key = move |item| item.id().get()
                let:item>
                <ItemView item />
            </For>
        </ul>
    }.into_any()
}

#[component]
fn ItemListView(
    items: KeyedSubfield<Store<Item>, Item, Uuid, Vec<Item>>
) -> impl IntoView
{
    view! {
        <ul class="items">
            <For
                each = move || items
                key = move |item| item.id().get()
                let:item>
                <ItemView item />
            </For>
        </ul>
    }.into_any()
}
```

they differ only by the first type argument to `KeyedSubfield<>`. try as i might, i can't find a way to unify these into one component. i can get as far as making the first type argument generic:
```rust
#[component]
fn TaskListView<SF>(
    #[allow(unused_variables)]
    tasks: KeyedSubfield<SF, Task, Uuid, Vec<Task>>
) -> impl IntoView
    where SF: StoreField<Value = Task> + Send + Sync
{
    view! { }
    // view! {
    //     <ul class="tasks">
    //         <For
    //             each = move || tasks
    //             key = move |task| task.id().get()
    //             let:task>
    //             <TaskView task />
    //         </For>
    //     </ul>
    // }.into_any()
}
```

but as soon as i uncomment out the `<For>`, i'm lost in a mess of not implementing `IntoIterator`, combinations of missing `Send` or `Sync`, etc.

would appreciate some guidance on how to structure this.

**Comment by gbj** [comment]
Sorry for the slow response. There's a straightforward solution here: I've implemented `Into<Field<T>>` for `Store<T>` in #3102, so you can simply do the following:
```rust
#[component]
fn ItemListView(
    items: KeyedSubfield<Field<Item>, Item, Uuid, Vec<Item>>
) -> impl IntoView
{
    view! {
        <ul class="items">
            <For
                each = move || items
                key = move |item| item.id().get()
                let:item>
                <ItemView item />
            </For>
        </ul>
    }.into_any()
}
```
(And the root call is something like `items=Field::from(store).children()`)

I'm pretty sure that works (I just checked against the example) but let me know if not. I decided passing the `KeyedSubfield` with its full type signature wasn't *too* bad, but in future it's also possible to create a type-erased version of that (a `KeyedField` just like there's `Field`), but it's kind of niche.

**Comment by wrl** [comment]
yeah that seems like pretty much exactly what i need! will await the merge of that branch and then i'll try porting my data structures over to stores again.

thanks :D
